src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -586,10 +613,22 @@ private void WarningRaised(object sender, BuildWarningEventArgs e)
                 threadId: e.ThreadId,
                 logOutputProperties: null);
 
+            if (ImmediateMessageRaised(message))
+            {
+                RenderImmediateMessage(message);
+            }
+
             project.AddBuildMessage(MessageSeverity.Warning, message);
         }
     }
 
+    /// <summary>
+    /// Detect markers that require special attention from a customer.
+    /// </summary>
+    /// <param name="message">Raised event.</param>
+    /// <returns>true if marker is detected.</returns>
+    private bool ImmediateMessageRaised(string message) => ImmediateMessageRegex().IsMatch(message);
```

Comment:
Naming nit: I'd prefer something like `IsImmediateMessage`; `ImmediateMessageRaised` sounds like an event handler to me.